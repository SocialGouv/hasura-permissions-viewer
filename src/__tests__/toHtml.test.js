const toHtml = require("..");

const permissions = require("./sample.json");

test("Should convert permissions to HTML", () => {
  expect(toHtml(permissions)).toMatchSnapshot();
});

console.log(toHtml(permissions));
