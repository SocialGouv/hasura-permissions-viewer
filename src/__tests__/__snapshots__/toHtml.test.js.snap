// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should convert metadata permissions to HTML 1`] = `
"
<style>.bg-danger-soft {background-color:#f7d4d7}</style>

    <table class=\\"table table-striped\\" style=\\"margin-bottom:50px\\">
      <thead class=\\"thead-dark\\">
        <tr>
          <th class=\\"thead-dark\\" colspan=\\"3\\">
            <h3>table books</h3>
          </th>
        </tr>
        <tr>
          <th scope=\\"col\\">Field</th>
          <th class=\\"text-center\\" scope=\\"col\\">anonymous</th>
<th class=\\"text-center\\" scope=\\"col\\">user</th>
        </tr>
      </thead>
      <tbody>
        <tr>
            <td>id</td>
            <td class=\\"text-center bg-danger-soft\\" key=\\"anonymous\\"><span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">R</span></td>
<td class=\\"text-center \\" key=\\"user\\"><span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">C</span>
<span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">R</span>
<span style=\\"cursor: pointer; color: red; -webkit-text-stroke: 1px #666;\\" title=\\"user_id:_eq:X-Hasura-User-Id\\">U</span>
<span style=\\"cursor: pointer; color: red; -webkit-text-stroke: 1px #666;\\" title=\\"user_id:_eq:X-Hasura-User-Id\\">D</span></td>
          </tr>
<tr>
            <td>name</td>
            <td class=\\"text-center bg-danger-soft\\" key=\\"anonymous\\"><span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">R</span></td>
<td class=\\"text-center \\" key=\\"user\\"><span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">C</span>
<span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">R</span>
<span style=\\"cursor: pointer; color: red; -webkit-text-stroke: 1px #666;\\" title=\\"user_id:_eq:X-Hasura-User-Id\\">U</span>
<span style=\\"cursor: pointer; color: red; -webkit-text-stroke: 1px #666;\\" title=\\"user_id:_eq:X-Hasura-User-Id\\">D</span></td>
          </tr>
<tr>
            <td>user_id</td>
            <td class=\\"text-center bg-danger-soft\\" key=\\"anonymous\\"><span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">R</span></td>
<td class=\\"text-center \\" key=\\"user\\"><span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">C</span>
<span style=\\"cursor: pointer; color: auto; -webkit-text-stroke: 1px #666;\\" title=\\"\\">R</span>
<span style=\\"cursor: pointer; color: red; -webkit-text-stroke: 1px #666;\\" title=\\"user_id:_eq:X-Hasura-User-Id\\">U</span>
<span style=\\"cursor: pointer; color: red; -webkit-text-stroke: 1px #666;\\" title=\\"user_id:_eq:X-Hasura-User-Id\\">D</span></td>
          </tr>
      </tbody>
      <thead class=\\"thead-dark\\">
        <tr>
          <th scope=\\"col\\">Filters</th>
          <th class=\\"text-center\\" scope=\\"col\\">anonymous</th>
<th class=\\"text-center\\" scope=\\"col\\">user</th>
        </tr>
      </thead>
       <tbody>
       <tr><td>Update</td>
         <td class=\\"text-center \\" key=\\"anonymous\\"></td>
<td class=\\"text-center \\" key=\\"user\\">user_id:_eq:X-Hasura-User-Id</td>
         </tr>
<tr><td>Delete</td>
         <td class=\\"text-center \\" key=\\"anonymous\\"></td>
<td class=\\"text-center \\" key=\\"user\\">user_id:_eq:X-Hasura-User-Id</td>
         </tr>
      </tbody>
       <thead class=\\"thead-dark\\">
        <tr>
          <th scope=\\"col\\">Sets</th>
          <th class=\\"text-center\\" scope=\\"col\\">anonymous</th>
<th class=\\"text-center\\" scope=\\"col\\">user</th>
        </tr>
      </thead>
       <tbody>
       <tr><td>Create</td>
         <td class=\\"text-center \\" key=\\"anonymous\\"></td>
<td class=\\"text-center \\" key=\\"user\\">user_id=x-hasura-User-id</td>
         </tr>
      </tbody>
    </table>"
`;
